codeland.html

```diff
@@ -549,34 +553,77 @@
 </style>
 
 <script>
-const DATA_URL = "/page/codeland_data";
-const REFRESH_RATE = 15000;
+const NOW_PLAYING_DATA_URL = "/page/codeland_now_playing";
+const NOW_PLAYING_REFRESH_RATE = 15000;
+const SCHEDULE_DATA_URL = "/page/codeland_schedule";
+const SCHEDULE_REFRESH_RATE = 120000;
+
+var scheduleData;
 
 const sleep = time => new Promise(resolve => setTimeout(resolve, time));
 const poll = (promiseFn, time) => promiseFn()
               .then(sleep(time)
               .then(() => poll(promiseFn, time)));
 
-const fetchData = url => (
+const fetchNowPlayingData = url => (
+  fetch(url)
+    .then((resp) => resp.json())
+    .then(function(data) {
+        updateTalkInfo(data.now_playing.talk_id);
+    })
+    .catch(function(error) {
+        console.log(error);
+    })
+);
+
+const fetchScheduleData = url => (
   fetch(url)
     .then((resp) => resp.json())
     .then(function(data) {
-        updateTalkInfo(data.talk);
+        scheduleData = data.schedule;
+        updateScheduleInfo(scheduleData);
     })
     .catch(function(error) {
         console.log(error);
     })
 );
 
 // Update the contents of a div with new data
-const updateTalkInfo = talkData => {
+const updateTalkInfo = talkId => {
+  talkData = scheduleData.find(element => element.id === talkId);
+
   document.querySelector('.codeland-livestream__info-title h2').textContent = `${talkData.title} by ${talkData.speaker}`;
   document.querySelector('.codeland-livestream__speaker-info div p').innerHTML = talkData.bio;
   document.querySelector('.codeland-livestream__speaker-info div button').dataset.url = talkData.url;
 };
 
+const updateScheduleInfo = scheduleData => {
+  document.querySelectorAll('.codeland-content__container .codeland-schedule__event').forEach(element => {
+    element.remove();
+  });
+  var listing_container = document.querySelector('.codeland-content__container');
+  var template = document.querySelector('#schedule-entry-template');
+  scheduleData.forEach(data => {
+    var start_time = new Date(Date.parse(data.start_time));
+    var end_time = new Date(Date.parse(data.end_time));
+
+    var listing = template.content.cloneNode(true);
+    listing.querySelector('.codeland-schedule__event').setAttribute("data-event-id", data.id);
+    listing.querySelector('.codeland-schedule__event-metadata').innerHTML = `${start_time.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' })} - ${end_time.toLocaleTimeString([], { hour: '2-digit', minute: '2-digit' })} `;
+    listing.querySelector('.codeland-schedule__event-info h3').innerHTML = data.speaker;
+    listing.querySelector('.codeland-schedule__event-title').innerHTML = data.title;
+    listing.querySelector('.codeland-schedule__event-speaker-bio').innerHTML = data.bio;
```

Comment:
The links from the schedule weren't working for me. I think we can add the `href` for the schedule link here, right? Something like:

```
listing.querySelector('a.codeland-schedule__event-link').href = data.url;
```

Reply:
Oh yes, absolutely!